### ISSUE TITLE
Simpler way to remove all log subscribers.

### ISSUE THREAD

**Issue by ioquatix** [roidrage/lograge#372]
Not sure if this has been considered, but what about:

```
ActiveSupport::Notifications.notifier = ActiveSupport::Notifications::Fanout.new
```

This clears out all subscriptions and is probably faster and more robust than the existing implementation.

**Comment by benlovell** [comment]
Nice. Thanks for the heads up. Do you want to open a PR or I can pick it up?

**Comment by ioquatix** [comment]
I was not planning on opening a PR but I can if that’s helpful.

**Comment by owst** [comment]
I agree that the existing implementation isn't particularly robust (see #386 for example), but wouldn't this also unsubscribe unrelated listeners, for example the one powering a gem such as https://github.com/steventen/sql_tracker
